### PR TITLE
update kzg lib to latest

### PR DESCRIPTION
**Pull request by roberto-bayardo**
also added a check that the length of the proofs array is as expected in validate_blob_transaction_wrapper per free the blobs

### CODE REVIEW

File: crypto/kzg/kzg.go [hunk 1]
```diff
@@ -77,7 +76,8 @@ func PointEvaluationPrecompile(input []byte) ([]byte, error) {
 	var versionedHash [32]byte
 	copy(versionedHash[:], input[:32])
 
-	var x, y [32]byte
+	var x gokzg4844.Scalar
```
Comment (kevaundray) [inline crypto/kzg/kzg.go:79]: x and y should be scalars here, so we can do 

var x,y gokzg4844.scalar

File: crypto/kzg/kzg.go [hunk 2]
```diff
@@ -86,12 +86,12 @@ func PointEvaluationPrecompile(input []byte) ([]byte, error) {
...
 	var dataKZG [48]byte
 	copy(dataKZG[:], input[96:144])
-	if KZGToVersionedHash(serialization.KZGCommitment(dataKZG)) != VersionedHash(versionedHash) {
+	if KZGToVersionedHash(gokzg4844.KZGCommitment(dataKZG)) != VersionedHash(versionedHash) {
 		return nil, errors.New("mismatched versioned hash")
 	}
 
 	// Quotient kzg: next 48 bytes
-	var quotientKZG [48]byte
+	var quotientKZG gokzg4844.Scalar
```
Comment (kevaundray) [inline crypto/kzg/kzg.go:94]: Here we want gokzg4844.KZGProof instead of Scalar

File: core/types/transaction_test.go [hunk 3]
```diff
@@ -493,10 +494,14 @@ func TestTransactionCoding(t *testing.T) {
...
 				},
 			}
+			cryptoCtx := kzg.CrpytoCtx()
+			blob := Blob{}
+			commitment, _ := cryptoCtx.BlobToKZGCommitment(gokzg4844.Blob(blob))
+			proof, _ := cryptoCtx.ComputeBlobKZGProof(gokzg4844.Blob(blob), commitment)
 			wrapData = &BlobTxWrapData{
-				BlobKzgs: BlobKzgs{KZGCommitment{0: 0xc0}},
-				Blobs:    Blobs{Blob{}},
-				Proofs:   KZGProofs{api.ZERO_POINT},
```
Comment (kevaundray) [inline core/types/transaction_test.go:499]: We changed api.ZERO_POINT to be gokzg4844.POINT_AT_INFINITY

**Review by kevaundray — COMMENTED** [review]
Left some comments re type changes :)

File: core/types/transaction_test.go (hunk 3 above)
Comment (roberto-bayardo) [reply core/types/transaction_test.go:499]: got it, though I think this version where we just appropriately compute the proofs/commitments for the test blob is better than exposing this.

File: crypto/kzg/kzg.go (hunk 2 above)
Comment (roberto-bayardo) [reply crypto/kzg/kzg.go]: ah I missed that the order of parameters in `VerifyKZGProof` has changed, that explains the failing test as well. all fixed.

**Review by kevaundray — APPROVED** [review]

**Pull request merged**